cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
Ok I understand. I think it would be nicer to have only one function that updates the solver object (like _populate solver or another one) and that only works with a reaction_list as this will make the lazy solver update easier as well. For instance the set_linear_coefficients code block in add_metabolites is pretty much the same that happens in _populate_solver only that is doesn't create the variables (but you coud add a check in populate_solver) and this happens in a lot of functions and will make it hard to find bugs or react to breaking changes in the optlang API.